sdk/samples/iot/paho_iot_hub_twin_cbor_sample.c

```diff
@@ -460,37 +461,56 @@ static bool parse_desired_device_count_property(
   bool property_found = false;
   *out_parsed_device_count = 0;
 
+  CborError rc; // CborNoError == 0
+
   // Parse message_span.
-  CborParser cbor_parser;
+  CborParser parser;
   CborValue root;
   CborValue desired_device_count;
 
-  // WARNING: Check the return of all API calls when developing your solution. Many Return checks
-  //          are ommited from this sample for simplification.
+  rc = cbor_parser_init((uint8_t*)message->payload, (size_t)message->payloadlen, 0, &parser, &root);
+  if (rc)
+  {
+    IOT_SAMPLE_LOG_ERROR("Failed to initiate parser: CborError %d.", rc);
+    exit(rc);
+  }
 
-  (void)cbor_parser_init(
-      (uint8_t*)message->payload, (size_t)message->payloadlen, 0, &cbor_parser, &root);
-  if (cbor_value_map_find_value(&root, desired_device_count_property_name, &desired_device_count)
-      == CborNoError)
+  rc = cbor_value_map_find_value(&root, desired_device_count_property_name, &desired_device_count);
+  if (rc)
   {
-    if (cbor_value_is_integer(&desired_device_count))
-    {
-      (void)cbor_value_get_int64(&desired_device_count, out_parsed_device_count);
-      property_found = true;
-    }
+    IOT_SAMPLE_LOG_ERROR(
+        "Error when searching for %s: CborError %d.", desired_device_count_property_name, rc);
+    exit(rc);
   }
 
-  if (property_found)
+  if (cbor_value_is_valid(&desired_device_count))
   {
-    IOT_SAMPLE_LOG(
-        "Parsed desired `%s`: %" PRIi64,
-        desired_device_count_property_name,
-        *out_parsed_device_count);
+    if (cbor_value_is_integer(&desired_device_count))
+    {
+      rc = cbor_value_get_int64(&desired_device_count, out_parsed_device_count);
+      if (rc)
+      {
+        IOT_SAMPLE_LOG_ERROR(
+            "Failed to get int64 value for %s: CborError %d.",
+            desired_device_count_property_name,
+            rc);
+        exit(rc);
+      }
+
+      IOT_SAMPLE_LOG(
+          "Parsed desired `%s`: %" PRIi64,
+          desired_device_count_property_name,
+          *out_parsed_device_count);
+    }
+    else
+    {
+      IOT_SAMPLE_LOG("`%s` property was not an integer", desired_device_count_property_name);
+    }
   }
   else
   {
     IOT_SAMPLE_LOG(
-        "`%s` property was not found in desired property response.",
+        "`%s` property was not found in desired property message.",
```

Comment:
sometimes there is a trailing period and something there isn't.

Reply:
I found a sentence that missed a period and have added it.  Other places don't have periods because they aren't sentences.